astroquery/xmm_newton/core.py

```diff
@@ -0,0 +1,343 @@
+
+            self._handler.get_file(filename, response=result, verbose=verbose)
+
+            return filename
+
+    def query_xsa_tap(self, query, output_file=None,
+                      output_format="votable", verbose=False):
+        """Launches a synchronous job to query the XSA tap
+
+        Parameters
+        ----------
+        query : str, mandatory
+            query (adql) to be executed
+        output_file : str, optional, default None
+            file name where the results are saved if dumpToFile is True.
+            If this parameter is not provided, the jobid is used instead
+        output_format : str, optional, default 'votable'
+            possible values 'votable' or 'csv'
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A table object
+        """
+
+        job = self._tap.launch_job(query=query, output_file=output_file,
+                                   output_format=output_format,
+                                   verbose=False,
+                                   dump_to_file=output_file is not None)
+        table = job.get_results()
+        return table
+
+    def get_tables(self, only_names=True, verbose=False):
+        """Get the available table in XSA TAP service
+
+        Parameters
+        ----------
+        only_names : bool, TAP+ only, optional, default 'False'
+            True to load table names only
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of tables
+        """
+
+        tables = self._tap.load_tables(only_names=only_names,
+                                       include_shared_tables=False,
+                                       verbose=verbose)
+        if only_names is True:
+            table_names = []
+            for t in tables:
+                table_names.append(t.name)
+            return table_names
+        else:
+            return tables
+
+    def get_columns(self, table_name, only_names=True, verbose=False):
+        """Get the available columns for a table in XSA TAP service
+
+        Parameters
+        ----------
+        table_name : string, mandatory, default None
+            table name of which, columns will be returned
+        only_names : bool, TAP+ only, optional, default 'False'
+            True to load table names only
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        A list of columns
+        """
+
+        tables = self._tap.load_tables(only_names=False,
+                                       include_shared_tables=False,
+                                       verbose=verbose)
+        columns = None
+        for t in tables:
+            if str(t.name) == str(table_name):
+                columns = t.columns
```

Comment:
We generally prefer variable names to be >1 character, though in this case it's a minor issue